equational-reasoning.md

```diff
@@ -0,0 +1,25 @@
+Equational reasoning
+====================
+
+**Tags**: *substitution model*, *referential transparency*, *purity*, *side effect*
+
+In terms of programming we can say that a function definition defines the scope where different expressions are evaluated in order to eventually return a result. This result defines the return type of that function.
```

Comment:
What about: `Concerning programming, we can say that a function definition limits the scope where various expressions are evaluated to produce a result. This result defines the return type of that function.`
And maybe should add something about the input of the function as well.